tests/virt_autotest/validate_system_health.pm

```diff
@@ -0,0 +1,84 @@
+# SUSE's openQA tests
+#
+# Copyright 2012-2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Do a basic examination to the host and guests after tests run
+# Maintainer: Julie CAO <JCao@suse.com>, qe-virt@suse.de
+
+use base "virt_feature_test_base";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use ipmi_backend_utils qw(reconnect_when_ssh_console_broken);
+use Utils::Architectures;
+use virt_autotest::common;
+use virt_autotest::utils qw(is_xen_host start_guests check_port_state check_host_health check_guest_health);
+use virt_utils qw(collect_host_and_guest_logs);
+
+sub prepare_run_test {
+    reconnect_when_ssh_console_broken unless defined(script_run("rm -f /root/{commands_history,commands_failure}", die_on_timeout => 0));
+    script_run("history -c");
+}
+
+sub run_test {
+
+    my $self = shift;
+    my %health_status = ();
+    $health_status{host} = check_host_health;
+    foreach my $guest (keys %virt_autotest::common::guests) {
+        script_run("virsh start $guest", die_on_timeout => 0);
+        if (script_retry("nmap $guest -PN -p ssh | grep open", delay => 2, retry => 60, die => 0) == 0) {
+            $health_status{$guest} = check_guest_health($guest);
+            if ($health_status{$guest} eq 'fail' || get_var('FORCE_UPLOAD_LOGS')) {
+                script_run("ssh root\@$guest 'dmesg > dmesg.txt 2>&1'", die_on_timeout => 0);
+            }
+        }
+        else {
+            $health_status{$guest} = "Unreachable";
+        }
+        record_info("Failed guest", "$guest: $health_status{$guest}", result => 'fail') unless $health_status{$guest} eq 'pass';
+    }
+    my @unhealthy_list = grep { $health_status{$_} ne 'pass' } keys %health_status;
+
+    # Upload logs on x86_64
+    if (is_x86_64 && (get_var('FORCE_UPLOAD_LOGS') || @unhealthy_list != 0)) {
+        script_run("tar cvzf libvirtd.tar.gz /var/log/libvirt", die_on_timeout => 0);
+        if (is_xen_host) {
+            script_run("tar cvzf xen_logs.tar.gz /var/log/xen", die_on_timeout => 0);
+            script_run("xl dmesg > xl_dmesg.txt", die_on_timeout => 0);
+        }
+        script_run("dmesg > dmesg_host.txt", die_on_timeout => 0);
+        my $host_extra_logs = "/root/libvirtd.tar.gz /root/dmesg_host.txt /root/xen_logs.tar.gz xl_dmesg.txt";
+        my $guest_extra_logs = "/root/dmesg.txt";
+        if (get_var('FORCE_UPLOAD_LOGS')) {
+            collect_host_and_guest_logs(join(' ', grep { $_ ne 'host' } keys %health_status), $host_extra_logs, $guest_extra_logs);
+        }
+        else {
+            collect_host_and_guest_logs(join(' ', grep { $_ ne 'host' } @unhealthy_list), $host_extra_logs, $guest_extra_logs);
+        }
+        $self->upload_coredumps;
```

Comment:
If you force upload coredumps please put in 
```
if (get_var('FORCE_UPLOAD_LOGS')) {
```
The logic here looks very confusing. Please help make the logic consistent.

Reply:
upload_coredump is needed if there either of unhealthy guests or FORCE_UPLOAD_LOGS.

Reply:
So I think the logic and variable used here little bit confusing.

Reply:
If heallthy status is good. Why do you need upload coredump ?

Reply:
I think 
```
if (get_var('FORCE_UPLOAD_LOGS')) {
            collect_host_and_guest_logs(join(' ', grep { $_ ne 'host' } keys %health_status), $host_extra_logs, $guest_extra_logs);
            $self->upload_coredumps;
}
```
is enough. Remove the else branch, because it will be collected anyway.

Reply:
No, without 'else collect_host_and_guest_logs`, the logs would not be collected if there are unhealthy host or guests.

Reply:
If there are unhealthy ones, test run will enter into post_fail_hook. There is collect_host_and_guest_logs in post_fail_hook already.

Reply:
Oh, I forgot collect_host_guest_logs were new added in post_fail_hook in this module. Then yes, you are right. I will remove 'else'.

Reply:
From "your logic"'s perspective, I think you just remove collect_host_guest_logs from post_fail_hook. Otherwise your whole code structure needs to refactor.

Reply:
oh, thanks. My brain has been thinking too much today, and it is getting blurred.